src/java/org/apache/cassandra/index/sai/disk/vector/CompactionGraph.java

```diff
@@ -476,7 +483,26 @@ public long ramBytesUsed()
 
     public boolean requiresFlush()
     {
-        return builder.getGraph().size() >= postingsEntriesAllocated;
+        return builder.getGraph().size() >= postingsEntriesAllocated || requiresFlush;
+    }
+
+    static <T> boolean safePut(ChronicleMap<T, CompactionVectorPostings> map, T key, CompactionVectorPostings value)
+    {
+        try
+        {
+            map.put(key, value);
+            return false;
+        }
+        catch (IllegalArgumentException e)
+        {
+            logger.debug("Error serializing postings to disk, will reattempt with compression", e);
+            // This is an extreme edge case where there are many duplicate vectors. This naive approach
+            // means that we might have a smaller vector graph than desired, but at least we will not
+            // fail to build the index.
+            value.setShouldCompress(true);
+            map.put(key, value);
```

Comment:
Is it possible it sill fails after compression? If so, then what? Maybe we should still provide at least some diagnostic message?